js/lex.go

```diff
@@ -161,6 +161,9 @@ func (l *Lexer) Next() (TokenType, []byte) {
 	// ErrorToken, WhitespaceToken and CommentToken are already returned
 	if tt == LineTerminatorToken || tt == PunctuatorToken && regexpStateByte[c] {
 		l.regexpState = true
+	} else if tt == IdentifierToken {
+		hash := ToHash(l.r.Lexeme())
+		l.regexpState = hash != 0 && hash != This
```

Comment:
I mean, yeah, we could list each of them explicitly, but "everything apart" seems like a nice base to start moving towards Sweet.js algorithm which does exactly that and so far seems to do great job for other lexers. Alternatively, yeah, we could make a table, and then we could even catch some obvious syntax errors (like `if` not followed by parentheses etc.).